set-api-deployment-hash/action.yml

```
@@ -0,0 +1,20 @@
+name: Set deployment hash
+description: Write SHA for current deployment to S3
+
+inputs:
+  s3-bucket:
+    description: S3 bucket
+    required: true
+  deployment-reference-file:
+    description: Filename to write the hash to and upload to S3
+    required: false
+    default: 'active-deployment-sha-${{ github.event.deployment.environment }}.txt'
+
+runs:
+  using: composite
+  steps:
+    - name: Update active deployment sha
+      run: |
+        echo -e "${{ github.sha }}" > ${{ inputs.deployment-reference-file }}
```

How do you feel about making "deployment-hash" an input as well? I understand that the current version works and have also check that `${{ github.sha }}` is resolved correctly, but I'm thinking that having this as an input would make it more explicit and easier to verify that the correct hash is used.

```suggestion
        echo -e "${{ inputs.deploylment-hash }}" > ${{ inputs.deployment-reference-file }}
```

---

If you argue like this, we should make `GITHUB_REPOSITORY` and `GITHUB_REPOSITORY_OWNER` an input as well.
If we pass it as an input, we have to copy and paste
```
with:
  deployment-hash: ${{ github.sha }}
  repository: ${{ github.repository }}
  repository-owner: ${{ github.repository_owner }}
```
in our `ci` and `deploy` workflow and for `insurance` and `profile`.

So for me it would be okay to keep it as it is, but we should update the naming.
So either use the `github` context like `${{ github.WHATEVER}}` or variables like `$GITHUB_WHATEVER`.

---

As discussed with @felix11h we don't need it as an input. There is no use-case so far to override the values.
We've decided to use the github context, so `GITHUB_REPOSITORY` will be updated to `${{ github.repository }}`, same for `GITHUB_REPOSITORY_OWNER`.